### DISCUSSION TITLE
Officially support redirecting to an external URL

### DISCUSSION THREAD

**Discussion by flavianh** [withastro/roadmap#847] [Proposal]
# Summary

At the moment, using 

```
export default defineConfig({
  ...
  redirects: {
    '/test2': {
      status: 302,
      destination: 'https://mysecondsite.com/test2',
    },
  },
});
```

redirects to `https://myfirstsite.com/https://mysecondsite.com/test2`, when it should redirect to the external URL directly.

This is something the Vercel CLI correctly warns about: `[WARN] [redirects] Redirecting to an external URL is not officially supported`.

# Background & Motivation

This is particularly useful to preserve links when migrating URLs into another domain or subdomain.

# Goals

My own case was:

1. Initially, I had a Ghost blog hosted on `mysite.com`
2. I moved it to `blog.mysite.com`, and `mysite.com` now hosts an Astro website. Links such as `mysite.com/article1` must redirect to `blog.mysite.com/article1`.

It is possible to use redirect lists in Cloudflare but this costs too much if the list of URLs is too large.
Redirecting with Astro allows to maintain this list more easily (as it can be contained in a plain text file such as a JSON or JS file).

**Comment by TheMikeyRoss** [comment]
Is this now possible?

**Reply by gxjansen** [reply]
On the current latest Astro 4.14.2 you'll still see the CLI warning and the redirect doesn't work (it'll return a 404)

**Reply by marty0678** [reply]
I'm using Astro 4.15.1 and I get the warning but it does appear to work and redirect to the other domain.

**Reply by mihailthebuilder** [reply]
I've been using this for a few weeks (on 4.15.9) and it works fine. Only just noticed the warning in my CLI ^_^

**Reply by ematipico** [reply]
Be all aware that you're leveraging a possible bug in Astro. This means that external redirects **aren't officially supported** and this could break your website.

**Comment by marcomonsanto** [comment]
Not using config redirects, but I was able to use `Response.redirect('URL')` inside the middleware to point to an external URL, could be an option if you can use middleware.

**Comment by JanTrichter** [comment]
I had a similar issue that I needed to go from my site's `/admin` to an external web page.

I solved it for now by adding an `admin.astro` file in the `/pages` directory and then redirect to the new website using JavaScript like this:

`/pages/admin.astro`
```html
<script>
    window.location.href = "https://website.com/whatever";
</script>
```

**Comment by ematipico** [comment]
I have a draft PR ready: https://github.com/withastro/astro/pull/12979

**Comment by ematipico** [comment]
Astro v5.2 ships with the support of external redirects! https://astro.build/blog/astro-520/

**Comment by flavianh** [comment]
Hi @ematipico , thank you for the work and follow-up!
I've tried it and I'm experiencing issues (Tested on Astro 5.3.0 and 5.4.1)

Here's my configuration:

```typescript
defineConfig({
  ...
  redirects: {
    "/author/": "https://blog.taktique.com/author/",
  }
})
```

Everything builds well, but the deployed (static) page `https://www.taktique.com/author/` erroneously redirects to `https://www.taktique.com/https:/blog.taktique.com/author/`.

See also this curl result

```bash
# curl -vvv https://www.taktique.com/author

* Request completely sent off
< HTTP/2 301 
< cache-control: public, max-age=0, must-revalidate
< content-type: text/plain
< date: Mon, 03 Mar 2025 08:56:42 GMT
< location: /https:/blog.taktique.com/author/
< server: Vercel
< x-vercel-id: xxxxx
< 
Redirecting...
```

**Reply by delucis** [reply]
Hi @flavianh — please report bugs with a minimal reproduction in the Astro repo: https://github.com/withastro/astro/issues/new/choose

Posts here might get missed and we request some specific information in issues to help us successfully debug.

**Reply by flavianh** [reply]
Good point I'll do this
